src/scripts/genEtwProvider.py

```diff
@@ -0,0 +1,323 @@
+##
+## Licensed to the .NET Foundation under one or more agreements.
+## The .NET Foundation licenses this file to you under the MIT license.
+## See the LICENSE file in the project root for more information.
+##
+## This script generates the interface to ETW using MC.exe
+##
+##  The intended use if for platforms which support event pipe
+##  but do not have a an eventing platform to recieve report ecents
+
+import os
+from os import path
+import shutil
+import re
+import sys
+import argparse
+import subprocess
+import xml.dom.minidom as DOM
+from genEventing import parseTemplateNodes
+
+xplatheader_filename = "clrxplatevents.h"
+macroheader_filename = "etwmacros.h"
+mcheader_filename = "ClrEtwAll.h"
+etw_dirname = "etw"
+replacements = [
+    (r"EventEnabled", "EventXplatEnabled"),
+    (r"\bPVOID\b", "void*")
+]
+
+stdprolog_cpp="""
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/******************************************************************
+
+DO NOT MODIFY. AUTOGENERATED FILE.
+This file is generated using the logic from <root>/src/scripts/genEtwProvider.py
+
+******************************************************************/
+"""
+stdprolog_cmake="""
+#
+#
+#******************************************************************
+
+#DO NOT MODIFY. AUTOGENERATED FILE.
+#This file is generated using the logic from <root>/src/scripts/genEtwProvider.py
+
+#******************************************************************
+"""
+
+def genProviderInterface(manifest, intermediate):
+    provider_dirname = os.path.join(intermediate, etw_dirname)
+
+    if not os.path.exists(provider_dirname):
+        os.makedirs(provider_dirname)
+
+    cmd = ['mc.exe', '-h', provider_dirname, '-r', provider_dirname, '-b', '-co', '-um', '-p', 'FireEtXplat', manifest]
```

Comment:
You should confirm that the codegen here is such that the FireEtXplat calls and EventEnabled calls for ETW inlined.  I suspect that they will be, but we should make sure we don't accidentally add call overhead.